salt/utils/pkg/deb.py

```diff
@@ -71,106 +453,307 @@ def repo_line(self):
-            opts_line,
-            uri,
-            dist,
-            " ".join(orig_comps),
-        ]
-        return SourceEntry(" ".join([line for line in repo_line if line.strip()]))
-
-    def remove(self, source):
-        """
-        remove a source from the list of sources
-        """
-        self.list.remove(source)
+            self.list.insert(pos, new_entry)
+        return new_entry
 
-    def save(self):
+    def remove(self, source_entry):
         """
-        write all of the sources from the list of sources
-        to the file.
+        Remove the entry from the sources.list
         """
-        filemap = {}
-        with tempfile.TemporaryDirectory() as tmpdir:
-            for source in self.list:
-                fname = pathlib.Path(tmpdir, pathlib.Path(source.file).name)
-                with salt.utils.files.fopen(str(fname), "a") as fp:
-                    fp.write(source.repo_line())
-                if source.file not in filemap:
-                    filemap[source.file] = {"tmp": fname}
+        self.list.remove(source_entry)
 
-            for fp in filemap:
-                shutil.move(str(filemap[fp]["tmp"]), fp)
+    def load_deb822_sections(self, file_obj):
+        """
+        Return Deb822 sections from .sources file object
+        """
+        sections = []
+        section = ""
+        for line in file_obj:
+            if not line.isspace():
+                # Consider not empty line as a part of a section
+                section += line
+            elif section:
+                # Add a new section on getting first space line
+                sections.append(Deb822Section(section))
+                section = ""
+
+        # Create the last section if we still have data for it
+        if section:
+            sections.append(Deb822Section(section))
+
+        return sections
+
+    def load(self, file_path):
+        """
+        Load the sources from the file
+        """
+        try:
+            with salt.utils.files.fopen(file_path) as f:
+                if file_path.endswith(".sources"):
+                    for section in self.load_deb822_sections(f):
+                        self.list.append(
+                            Deb822SourceEntry(section, file_path, list=self)
+                        )
+                else:
+                    for line in f:
+                        source = SourceEntry(line, file_path)
+                        self.list.append(source)
+        except Exception as exc:  # pylint: disable=broad-except
+            log.error("Could not parse source file '%s'", file_path, exc_info=True)
+
+    def index(self, entry):
+        return self.list.index(entry)
+
+    def save(self):
+        """save the current sources"""
+        # write an empty default config file if there aren't any sources
+        if len(self.list) == 0:
+            path = _APT_SOURCES_LIST
+            header = (
+                "## See sources.list(5) for more information, especialy\n"
```

Comment:
typo `especially`. Especially what? The next line starts a new sentence (capital R `Remember`)

Reply:
It's default header used by `apt`, you can find it in `aptsources/sources.py` in https://launchpad.net/python-apt